src/ipahealthcheck/ipa/certs.py

```diff
@@ -583,6 +584,103 @@ def check(self):
+            except errors.NotFound:
+                yield Result(plugin, constants.ERROR,
+                             msg='%s entry not found in LDAP' % cert_dn)
+                return False
+            try:
+                nsscert = db.get_cert_from_db(cert_nick)
+            except Exception as e:
+                yield Result(plugin, constants.ERROR,
+                             error=str(e),
+                             msg=('Unable to load %s certificate:'
+                                  '{error}' % cert_nick))
+                return False
+            ldapcert = entry[attr][0]
+            if nsscert != ldapcert:
+                yield Result(plugin, constants.ERROR,
+                             msg=('\'%s\' certificate in NSS DB does not '
+                                  'does not match entry in LDAP' % cert_nick))
+                return False
+            return True
+
+        # Ensure /etc/ipa/ca.crt matches NSS DB CA certificates
+        def match_cacert_and_db(plugin, cacert, dbpath):
+            db = certs.CertDB(api.env.realm, dbpath)
+            try:
+                dbcacert = db.get_cert_from_db('%s IPA CA' % api.env.realm)
+            except Exception as e:
+                yield Result(plugin, constants.ERROR,
+                             error=str(e),
+                             msg='Unable to load CA cert: {error}')
+                return False
+            if cacert != dbcacert:
+                yield Result(plugin, constants.ERROR,
+                             msg=('CA Certificate from %s does not match %s'
+                                  % (dbpath, paths.IPA_CA_CRT)))
+                return False
+            return True
+
+        db = certs.CertDB(api.env.realm, paths.PKI_TOMCAT_ALIAS_DIR)
+        dn = DN('uid=pkidbuser,ou=people,o=ipaca')
+        nick = 'subsystemCert cert-pki-ca'
+        subsystem_ok = yield from match_ldap_nss_cert(self, self.conn,
+                                                      db, dn,
+                                                      'userCertificate', nick)
+        dn = DN('cn=%s IPA CA' % api.env.realm,
+                'cn=certificates,cn=ipa,cn=etc',
+                api.env.basedn)
+        nick = 'caSigningCert cert-pki-ca'
+        casigning_ok = yield from match_ldap_nss_cert(self, self.conn,
+                                                      db, dn, 'CACertificate',
+                                                      nick)
+
+        try:
+            cacert = x509.load_certificate_from_file(paths.IPA_CA_CRT)
+        except Exception as e:
+            yield Result(self, constants.ERROR,
+                         error=str(e),
+                         msg='Unable to load CA cert: {error}')
+            return
+
+        # Ensure CA cert entry from LDAP matches /etc/ipa/ca.crt
+        try:
+            entry = self.conn.get_entry(DN('cn=CAcert,cn=ipa,cn=etc',
+                                           api.env.basedn))
```

Comment:
This should still be valid but it is the legacy location for storing the CA certificate. It lacks the full chain so may not be all that useful by itself.

The full chain is stored in cn=certificates,cn=ipa,cn=etc,$SUFFIX.

I suspect that with an external IPA install the check against ca.crt will fail.